go/vt/topo/helpers/copy.go

```diff
@@ -37,44 +33,33 @@ func CopyKeyspaces(ctx context.Context, fromTS, toTS *topo.Server) {
 		log.Fatalf("GetKeyspaces: %v", err)
 	}
 
-	wg := sync.WaitGroup{}
-	rec := concurrency.AllErrorRecorder{}
 	for _, keyspace := range keyspaces {
-		wg.Add(1)
-		go func(keyspace string) {
-			defer wg.Done()
 
-			ki, err := fromTS.GetKeyspace(ctx, keyspace)
-			if err != nil {
-				rec.RecordError(fmt.Errorf("GetKeyspace(%v): %v", keyspace, err))
-				return
-			}
+		ki, err := fromTS.GetKeyspace(ctx, keyspace)
+		if err != nil {
+			log.Errorf("GetKeyspace(%v): %v", keyspace, err)
+			continue
+		}
 
-			if err := toTS.CreateKeyspace(ctx, keyspace, ki.Keyspace); err != nil {
-				if topo.IsErrType(err, topo.NodeExists) {
-					log.Warningf("keyspace %v already exists", keyspace)
-				} else {
-					rec.RecordError(fmt.Errorf("CreateKeyspace(%v): %v", keyspace, err))
-				}
+		if err := toTS.CreateKeyspace(ctx, keyspace, ki.Keyspace); err != nil {
+			if topo.IsErrType(err, topo.NodeExists) {
+				log.Warningf("keyspace %v already exists", keyspace)
+			} else {
+				log.Errorf("CreateKeyspace(%v): %v", keyspace, err)
 			}
+		}
 
-			vs, err := fromTS.GetVSchema(ctx, keyspace)
-			switch {
-			case err == nil:
-				if err := toTS.SaveVSchema(ctx, keyspace, vs); err != nil {
-					rec.RecordError(fmt.Errorf("SaveVSchema(%v): %v", keyspace, err))
-				}
-			case topo.IsErrType(err, topo.NoNode):
-				// Nothing to do.
-			default:
-				rec.RecordError(fmt.Errorf("GetVSchema(%v): %v", keyspace, err))
-				return
+		vs, err := fromTS.GetVSchema(ctx, keyspace)
+		switch {
+		case err == nil:
+			if err := toTS.SaveVSchema(ctx, keyspace, vs); err != nil {
+				log.Errorf("SaveVSchema(%v): %v", keyspace, err)
 			}
-		}(keyspace)
-	}
-	wg.Wait()
-	if rec.HasErrors() {
```

Comment:
Looks like the refactor removed this log.Fatalf if any of the operations failed and instead will just log each individual error.

I think that’s undesirable and that instead we should track the individual errors and fatal the operation if any occurred

Reply:
Alternatively we could just fatal if any error occurs mid copy.